src/libraries/Native/Unix/System.Security.Cryptography.Native.Android/pal_cipher.c

```diff
@@ -0,0 +1,363 @@
+
+    if (type == AndroidCryptoNative_Aes192Ecb())    return JSTRING("AES/ECB/NoPadding");
+    if (type == AndroidCryptoNative_Aes192Cbc())    return JSTRING("AES/CBC/NoPadding");
+    if (type == AndroidCryptoNative_Aes192Gcm())    return JSTRING("AES/GCM/NoPadding");
+    if (type == AndroidCryptoNative_Aes192Ccm())    return JSTRING("AES/CCM/NoPadding");
+    if (type == AndroidCryptoNative_Aes192Cfb8())   return JSTRING("AES/CFB/NoPadding");
+
+    if (type == AndroidCryptoNative_Aes256Ecb())    return JSTRING("AES/ECB/NoPadding");
+    if (type == AndroidCryptoNative_Aes256Cbc())    return JSTRING("AES/CBC/NoPadding");
+    if (type == AndroidCryptoNative_Aes256Gcm())    return JSTRING("AES/GCM/NoPadding");
+    if (type == AndroidCryptoNative_Aes256Ccm())    return JSTRING("AES/CCM/NoPadding");
+    if (type == AndroidCryptoNative_Aes256Cfb8())   return JSTRING("AES/CFB/NoPadding");
+
+    if (type == AndroidCryptoNative_DesEcb())       return JSTRING("DES/ECB/NoPadding");
+    if (type == AndroidCryptoNative_DesCfb8())      return JSTRING("DES/CFB/NoPadding");
+    if (type == AndroidCryptoNative_DesCbc())       return JSTRING("DES/CBC/NoPadding");
+
+    if (type == AndroidCryptoNative_Des3Ecb())      return JSTRING("DESede/ECB/NoPadding");
+    if (type == AndroidCryptoNative_Des3Cbc())      return JSTRING("DESede/CBC/NoPadding");
+    if (type == AndroidCryptoNative_Des3Cfb8())     return JSTRING("DESede/CFB/NoPadding");
+    if (type == AndroidCryptoNative_Des3Cfb64())    return JSTRING("DESede/CFB/NoPadding");
+
+    if (type == AndroidCryptoNative_Aes128Cfb128()) return JSTRING("AES/CFB128/NoPadding");
+    if (type == AndroidCryptoNative_Aes192Cfb128()) return JSTRING("AES/CFB128/NoPadding");
+    if (type == AndroidCryptoNative_Aes256Cfb128()) return JSTRING("AES/CFB128/NoPadding");
+
+    LOG_ERROR("This algorithm (%ld) is not supported", (long)type);
+    return FAIL;
+}
+
+static bool HasTag(intptr_t type)
+{
+    return (type == AndroidCryptoNative_Aes128Gcm()) ||
+           (type == AndroidCryptoNative_Aes128Ccm()) ||
+           (type == AndroidCryptoNative_Aes192Gcm()) ||
+           (type == AndroidCryptoNative_Aes192Ccm()) ||
+           (type == AndroidCryptoNative_Aes256Gcm()) ||
+           (type == AndroidCryptoNative_Aes256Ccm());
+}
+
+CipherCtx* AndroidCryptoNative_CipherCreatePartial(intptr_t type)
+{
+    JNIEnv* env = GetJNIEnv();
+    jobject algName = GetAlgorithmName(env, type);
+    if (!algName)
+        return FAIL;
+
+    jobject cipher = ToGRef(env, (*env)->CallStaticObjectMethod(env, g_cipherClass, g_cipherGetInstanceMethod, algName));
+    (*env)->DeleteLocalRef(env, algName);
+
+    CipherCtx* ctx = malloc(sizeof(CipherCtx));
+    ctx->cipher = cipher;
+    ctx->type = type;
+    ctx->tagLength = TAG_MAX_LENGTH;
+    ctx->ivLength = 0;
+    ctx->encMode = 0;
+    ctx->key = NULL;
+    ctx->iv = NULL;
+    return CheckJNIExceptions(env) ? FAIL : ctx;
```

Comment:
Seems like we can check this above - prior to the malloc. In this case I assume we will have a memory leak now since we indicated success.